spec/mongoid/association/referenced/belongs_to/buildable_spec.rb

```diff
@@ -14,31 +14,70 @@
       association.build(base, object)
     end
 
+    let(:options) do
+      {}
+    end
+
     let(:association) do
-      Post.relations['person']
+      Mongoid::Association::Referenced::BelongsTo.new(Post, :person, options)
     end
 
     context "when provided an id" do
 
       context "when the object is an object id" do
 
         let!(:person) do
-          Person.create(_id: object)
+          Person.create(_id: object, username: 'Bob')
         end
 
         let(:object) do
           BSON::ObjectId.new
         end
 
         before do
-          expect(Person).to receive(:where).with(association.primary_key => object).and_call_original
+          expect_any_instance_of(Mongoid::Criteria).to receive(:where).with(association.primary_key => object).and_call_original
```

Comment:
What is the purpose of this and similar changes throughout the PR please?

Reply:
It's related to this change in buildable:

```ruby
          def query_criteria(object, base)
            crit = klass.criteria                        # <== this line
            crit = crit.apply_scope(scope)
            crit = crit.where(foreign_key => object)
            with_polymorphic_criterion(crit, base)
          end
```

We're now explicitly creating the criteria first line, so the call doesn't happen on the Person class, it happens on a Criteria object.